micrometer-core/src/main/java11/io/micrometer/core/instrument/binder/jdk/DefaultHttpClientObservationConvention.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.jdk;
+
+import io.micrometer.common.KeyValues;
+import io.micrometer.common.lang.Nullable;
+
+import java.net.http.HttpRequest;
+
+/**
+ * Default implementation of {@link HttpClientObservationConvention}.
+ *
+ * @author Marcin Grzejszczak
+ * @since 1.10.0
+ */
+public class DefaultHttpClientObservationConvention implements HttpClientObservationConvention {
+
+    @Override
+    public KeyValues getLowCardinalityKeyValues(HttpClientContext context) {
+        if (context.getCarrier() == null) {
+            return KeyValues.empty();
+        }
+        HttpRequest httpRequest = context.getCarrier().build();
+        KeyValues keyValues = KeyValues.of(
+                HttpClientDocumentedObservation.LowCardinalityKeys.METHOD.withValue(httpRequest.method()),
+                HttpClientDocumentedObservation.LowCardinalityKeys.URI.withValue(httpRequest.uri().toString()));
```

Comment:
We probably want some way for users of HttpClient to specify a templated URI. Or open a follow-up issue for this, since some other http client instrumentation has potentially high cardinality uri tag by default.

Reply:
> We probably want some way for users of HttpClient to specify a templated URI. Or open a follow-up issue for this, since some other http client instrumentation has potentially high cardinality uri tag by default.

I couldn't find a way to pass a templated URI using `HttpClient`. Do you have an example of how to do that?

Reply:
Looks like this is resolved with the `uriMapper` introduced